.github/workflows/nightly-benchmark.yml

```diff
@@ -0,0 +1,71 @@
+name: Nightly Gas Costs Benchmark
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+    inputs:
+      create_pr:
+        description: "Create a pull request with the benchmark results"
+        required: true
+        default: "false"
+
+env:
+  GIT_REPO: ${{ github.repository }}
+  GIT_REPO_NAME: ${{ github.event.repository.name }}
+  CARGO_TERM_COLOR: always
+  RUST_VERSION: 1.81.0
+  CONSENSUS_PARAMETERS_VERSION: V2
+
+jobs:
+  benchmark:
+    runs-on:
+      group: fuelcore-benchmark
+    outputs:
+      benchmark_results: ${{ steps.benchmarks.outputs.benchmark_results }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Install Rust Toolchain
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ env.RUST_VERSION }}
+          targets: "wasm32-unknown-unknown"
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+      - name: Run benchmarks
+        id: benchmarks
+        run: |
+          cargo criterion -p fuel-core-benches --message-format json --bench vm > nightly_benchmarks.json
+          cargo run -p fuel-core-benches --bin collect --release -- --input nightly_benchmarks.json --format consensus-parameters --output nightly_benchmarks.json
+          echo "benchmark_results=$(cat nightly_benchmarks.json | tr -d '\n' | jq '.${{ env.CONSENSUS_PARAMETERS_VERSION }}.gas_costs')" # this will fail when consensus parameters version changes
+  create_pr:
+    needs: benchmark
+    runs-on:
+      group: fuelcore-benchmark
```

Comment:
Don't think this needs to run on the benchmark server - I'd recommend leaving this as `runs-on: ubuntu-latest` so we keep the benchmark server as clean as possible.

Reply:
addressed in ec9877b150